ll2_global_planner/src/ll2_global_planner_nodelet.cpp

```diff
@@ -44,10 +49,178 @@ void Ll2GlobalPlannerNl::loadParams()
 
 void Ll2GlobalPlannerNl::laneletMapCb(const autoware_lanelet2_msgs::MapBin& map_msg)
 {
+  lanelet_map_ = std::make_shared<lanelet::LaneletMap>();
+  lanelet::utils::conversion::fromBinMsg(map_msg, lanelet_map_);
+  initialized_ = true;
+  ROS_INFO("Loaded Lanelet map");
 }
 
 void Ll2GlobalPlannerNl::poseGoalCb(const geometry_msgs::PoseStamped::ConstPtr& pose_msg)
 {
+  if (!initialized_)
+  {
+    ROS_WARN("Waiting for lanelet2 map");
+    return;
+  }
+
+  planRoute(pose_msg->pose.position);
+}
+
+void Ll2GlobalPlannerNl::planRoute(const geometry_msgs::Point& goal_point)
+{
+  const float duplicate_dist_threshold = 0.1;
+
+  // Find the nearest lanelet to the goal point
+  BasicPoint2d local_point(goal_point.x, goal_point.y);
+  Lanelet goal_lanelet = getNearestLanelet(local_point);
+
+  // Get the current vehicle position
+  geometry_msgs::TransformStamped tf_msg;
+  try
+  {
+    tf_msg = tf_buffer_.lookupTransform("map", "base_link", ros::Time(0), ros::Duration(2.0));
+  }
+  catch (tf2::TransformException &ex)
+  {
+    ROS_WARN_THROTTLE(2, "%s", ex.what());
+    ROS_WARN_THROTTLE(2, "Waiting for map -> base_link transform");
+    return;
+  }
+
+  BasicPoint2d starting_point(
+    tf_msg.transform.translation.x, tf_msg.transform.translation.y);
+  const Lanelet& starting_lanelet = getNearestLanelet(starting_point);
+
+  // Plan a route from current vehicle position
+  traffic_rules::TrafficRulesPtr traffic_rules{traffic_rules::TrafficRulesFactory::instance().create(Locations::Germany, Participants::Vehicle)};
```

Comment:
Should the routing graph be called everytime we plan a route?
The graph should only change when the map changes, in this case, only when recieved from the map provider.
This could cut down on planning time significantly.

Reply:
Good call! This should only be done in the `laneletMapCb`.